SPECS-EXTENDED/rubygem-coderay/rubygem-coderay.spec

```diff
@@ -35,53 +30,44 @@ BuildArch: noarch
 Documentation for %{name}.
 
 %prep
-%setup -q -n %{gem_name}-%{version}
-
-%patch0 -p1
+%autosetup -p1 -n %{gem_name}-%{version}
 
 %build
-gem build ../%{gem_name}-%{version}.gemspec
-
-%gem_install
+gem build %{gem_name}
 
 %install
-mkdir -p %{buildroot}%{gem_dir}
-cp -a .%{gem_dir}/* \
-        %{buildroot}%{gem_dir}/
-
-
-mkdir -p %{buildroot}%{_bindir}
-cp -a .%{_bindir}/* \
-        %{buildroot}%{_bindir}/
-
-find %{buildroot}%{gem_instdir}/bin -type f | xargs chmod a+x
+gem install -V --local --force --install-dir %{buildroot}/%{gemdir} %{gem_name}-%{version}.rc1.gem
+mkdir -p %{buildroot}%{gem_instdir}
+#add lib and bin files to buildroot from Source0
+cp -a lib/ %{buildroot}%{gem_instdir}/
+cp -a bin/ %{buildroot}%{gem_instdir}/
+#add MIT-LICENSE file to buildroot from Source0
+cp MIT-LICENSE %{buildroot}%{gem_instdir}/
+cp README_INDEX.rdoc %{buildroot}%{gem_instdir}/
 
 %check
-pushd .%{gem_instdir}
-tar xzvf %{SOURCE1}
-# See https://github.com/rubychan/coderay/blob/master/rake_tasks/test.rake
 LANG=C.UTF-8
 ruby ./test/functional/suite.rb
 ruby ./test/functional/for_redcloth.rb
 ruby ./test/unit/suite.rb
-popd
 
 %files
 %dir %{gem_instdir}
-%{_bindir}/coderay
 %license %{gem_instdir}/MIT-LICENSE
 %{gem_instdir}/bin
 %{gem_libdir}
-%exclude %{gem_cache}
-%{gem_spec}
+%exclude /%{gemdir}/cache
+/%{gemdir}/specifications
 
 %files doc
-%doc %{gem_docdir}
+%doc /%{gemdir}/doc/
 %doc %{gem_instdir}/README_INDEX.rdoc
+/%{gemdir}/gems/%{gem_name}-%{version}.rc1/README_INDEX.rdoc
```

Comment:
This looks suspicious. If it's a documentation file, then let's add it with the `%doc` macro. But it also looks similar to the file one line above it. Are we sure this is needed?